### PR TITLE
SDK and Build tools are updated

### PR DESCRIPTION
**Pull request by tasomaniac**
I couldn't wait :)
Support library will also be needed to be updated soon. (I think)

### CODE REVIEW

**Comment by tasomaniac** [comment]
With the new build-tool it is the lint failing with 345 issues found

**Comment by stefanhoth** [comment]
Ha, you beat me to it (but only because I added this first on a work project :wink: )

**Comment by stefanhoth** [comment]
I'll have a look at the issues

**Comment by stefanhoth** [comment]
Lol, what a fail. :-1: 

image

How about releasing this before adding this lint check as an error?

**Comment by stefanhoth** [comment]
Btw, I only see
```
Check performed at Wed Mar 11 00:10:06 CET 2015.
2 errors and 274 warnings found:
```

**Comment by stefanhoth** [comment]
Sooo, with my lint adjustment from ccc6d95 the build passes but according to the message above it's not recommended to work like that. What do you think, @friedger @HenriqueRocha @tasomaniac?

Shall we just go for it anyway and play with it?

**Comment by tasomaniac** [comment]
I think if we wait for the new support library to come out it will be fine.

**Comment by tasomaniac** [comment]
I've just saw the Support version 22 changelog. http://developer.android.com/tools/support-library/index.html
It is about to be released. And guess what, we have AppCompat Buttons now. Yey!

**Comment by stefanhoth** [comment]
:+1:

**Comment by stefanhoth** [comment]
@tasomaniac The support libs dropped so I updated the PR. Hope the build will be fine now. :smile:

**Comment by tasomaniac** [comment]
Finally! :)

**Comment by tasomaniac** [comment]
Rebased for merge conflicts. 
Can you @all review this again? I've upgraded the other libraries as well. All of them are minor updates without an API change. 
And removed unused 2 google libraries. The application seems to be fine. Please build yourself and check before merge.

File: app/build.gradle [hunk 1]
```diff
@@ -145,15 +145,13 @@ dependencies {
     compile 'com.google.android.gms:play-services-appstate:6.5.87'
 
     //Google
-    compile 'com.google.code.gson:gson:2.3'
+    compile 'com.google.code.gson:gson:2.3.1'
     compile 'com.google:volley:4.3'
-    compile 'com.google.http-client:google-http-client-gson:1.18.0-rc'
+    compile 'com.google.http-client:google-http-client-gson:1.19.0'
     compile 'com.google.apis:google-api-services-plus:v1-rev126-1.18.0-rc'
```
Comment (stefanhoth) [inline app/build.gradle]: @tasomaniac do you know what this is for?

Comment (tasomaniac) [inline app/build.gradle]: Which one? Http-client? 
I want to get rid of it and use joyful Retrofit everywhere. Right now, Lots of API calls depends on `google-http-client-gson` unfortunately.

Comment (stefanhoth) [inline app/build.gradle]: That one `com.google.apis:google-api-services-plus:v1-rev126-1.18.0-rc`

Comment (tasomaniac) [inline app/build.gradle]: I also wanted to delete it but it is used. When I upgrade it to version `1.19.0` it exceeds the 65K dex method count.

Comment (stefanhoth) [inline app/build.gradle]: I was just curious what's in there what is not part of GMS.

**Comment by stefanhoth** [comment]
Great job. :+1: 

:shipit:

**Pull request merged**